qiskit_machine_learning/algorithms/serializable_model.py

```diff
@@ -0,0 +1,62 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""A mixin class for saving and loading models."""
+
+import logging
+from typing import Any
+
+import dill
+
+
+logger = logging.getLogger(__name__)
+
+
+class SerializableModelMixin:
+    """
+    This class provides convenient methods for saving and loading models.
+    """
+
+    def save(self, file_name: str) -> None:
+        """
+        Saves this model to the file. Internally, the model is serialized via ``dill`` and saved to
+        the specified file. All parameters are saved, including a quantum instance that is
+        referenced by internal objects. That means if a model is loaded from a model and is used
+        the same quantum instance and a corresponding backend will be used.
```

Comment:
> The second sentence pretty much just repeats at the end what was said initially. How about `` Saves this model to the specified file. Internally, the model is serialized via ``dill``. `` I assume we are exposing internals to inform the user better on the file etc and the fact that they can say use dill global settings if needed to override the dump defaulted parameters that are not supplied explicitly here?

Updated the docstring.
Yes, if a user wants to tweak it or load manually or anything like this.

> 
> > All parameters are saved, including a quantum instance that is referenced by internal objects.
> 
> By all parameters you mean the object state since these parameters are stored in the object in some way?
> 

This is correct. I prefer not to use object state, it is less clear than just "all parameters" for less experienced users. If you think you can re-phrased it better, I'm totally fine.

> > That means if a model is loaded from a model and is used  the same quantum instance and a corresponding backend will be used.
> 
> if a model is loaded from a ~model~ file ? (Though of course presumably update the quantum instance etc if they want)

Edited.